helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/clusterRole.yaml

```diff
@@ -10,7 +10,7 @@ metadata:
     heritage: {{ .Release.Service }}
 rules:
 {{- if .Values.podSecurityPolicy.create }}
-- apiGroups:      ['extensions']
+- apiGroups:      ['policy']
```

Comment:
it will mean that for those who are on <1.10, it will break, yeah? 
Can we make it configurable from values.yaml file? 
is there any best practices for this?

Reply:
Is there an explicit list of Kubernetes versions that Splunk Connect for Kubernetes supports?  1.10 is quite old - I wonder if all other features used here are present/functional.  Helm 3 commits to n-3 support.  A user would need to go all the way back to Helm 2.11.x to have Kubernetes 1.10 support.
https://helm.sh/docs/topics/version_skew/

Kubernetes itself supports only the three most recent minor releases (1.18, 1.19, and 1.20 as I write this).
https://kubernetes.io/docs/setup/release/version-skew-policy/

Helm documentation has an entire page dedicated to Deprecated Kubernetes APIs:
https://helm.sh/docs/topics/kubernetes_apis/

Reply:
At least OpenShift 3.11 (K8s 1.11) is still very common among enterprises. 

Helm 3 works fine with openshift 3.11 as long as you use `--disable-openapi-validation` helm option (openshift 3.11 ships with schema definitions which mark fields as required which are not required in every situation).

Reply:
We could certainly allow override with a `podSecurityPolicy.apiGroup` parameter in values.yaml.  I would suggest we default to "policy" in the interest of forward compatibility - that should work even with OpenShift 3.11.

Reply:
we don't have the list of k8s versions we support.
yea I agree with your suggestion to allow override but make `policy` as default.

Reply:
added podSecurityPolicy.apiGroup parameter in latest commits (b88c519).